### PR TITLE
Singleton.instance() now accepts parameters

### PR DESCRIPTION
**Pull request by mariushegele**
Speeds up unit tests

### CODE REVIEW

File: preferences.json [hunk 1]
```diff
@@ -17,20 +17,22 @@
 		"mainstation_coords": [48.782761, 9.182630],
 		"home_coords": [48.777980, 9.181840]
 	},
-    "cal": {
+        "cal": {
```
Comment (Ovakefali13) [inline preferences.json]: Why the tabs here?

**Comment by Ovakefali13** [comment]
Crap, it only saved one comment... Well, to sum up the comments, I mainly looked into the files I created. But I guess you already fixed these issues

**Comment by Ovakefali13** [comment]
So if the tests run through I guess we can merge into develop

**Comment by mariushegele** [comment]
Thanks!

**Pull request merged**